bundles/org.openhab.binding.argoclima/src/main/java/org/openhab/binding/argoclima/internal/handler/ArgoClimaHandlerBase.java

```diff
@@ -0,0 +1,829 @@
+            throws ArgoRemoteServerStubStartupException, ArgoConfigurationException;
+
+    /**
+     * {@inheritDoc}
+     *
+     * @implNote Initializes thing config and device API, and continues the thing initialization asynchronously through
+     *           {@link #initializeThing()} - as this method must return quickly. Also launches device state regular
+     *           polling (if configured) -
+     *           {@link #startAutomaticRefresh()}. While the poll will also (re)initialize the device, a dedicated
+     *           initialization logic is kept b/c polling may be disabled by the user (and there's no harm in triggering
+     *           both poll and refresh -> first to complete will win).
+     * @implNote If either of the initialize/poll threads are launched, both {@link #config} and {@link #deviceApi} are
+     *           guaranteed to have values (so their use is safe from any other method from this class except for
+     *           {@code dispose()}, as they are either invoked by the threads started herein, or guaranteed by the
+     *           framework to not get called if the device is not initialized. Hence a check for successful
+     *           initialization is NOT performed on each and every method.
+     */
+    @Override
+    public final void initialize() {
+        // Step0: If this a re-initialize (ex. config change), let's stop everything and start anew (not supporting
+        // graceful updates to the refresher threads and/or passthrough server)
+        this.config.ifPresent(c -> stopRunningTasks());
+
+        // Step1: Init config
+        ConfigT config;
+        try {
+            config = getConfigInternal();
+            this.config = Optional.of(config);
+        } catch (ArgoConfigurationException ex) {
+            logger.warn("[{}] {}", getThing().getUID().getId(), ex.getMessage());
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, ex.getLocalizedMessage());
+            return;
+        }
+        logger.debug("[{}] Running with config: {}", getThing().getUID(), config.toString());
+
+        var configValidationError = config.validate();
+        if (!configValidationError.isEmpty()) {
+            var message = i18nProvider.getText("thing-status.argoclima.invalid-config",
+                    "Invalid thing configuration. {0}", configValidationError);
+            logger.warn("[{}] {}", getThing().getUID(), message);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, message);
+            return;
+        }
+
+        // Step2: Init device API (this will start passthrough server & client threads, if configured)
+        try {
+            this.deviceApi = Optional.of(initializeDeviceApi(config));
+        } catch (ArgoRemoteServerStubStartupException | ArgoConfigurationException e) {
+            logger.warn("[{}] Failed to initialize Device API. Error: {}", getThing().getUID(), e.getMessage());
```

Comment:
See previous comment about logging and setting thing status in the same block. Please adjust all occurences in all files.